File: frontend/src/types/room.ts
// Mirrors app/schemas/room.py exactly.

export interface Room {
  id: number;
  room_name: string;
  room_number: string | null;
  room_type: string | null;
  floor_id: number;
  area: number | null;
  occupancy: number | null;
}

export interface RoomCreateInput {
  room_name: string;
  room_number?: string | null;
  room_type?: string | null;
  floor_id: number;
  area?: number | null;
  occupancy?: number | null;
}

// RoomUpdate has no floor_id — a room cannot be moved between floors via this API.
export interface RoomUpdateInput {
  room_name?: string;
  room_number?: string | null;
  room_type?: string | null;
  area?: number | null;
  occupancy?: number | null;
}

File: backend/app/schemas/room.py
from pydantic import BaseModel, ConfigDict


class RoomCreate(BaseModel):
    room_name: str
    room_number: str | None = None
    room_type: str | None = None
    floor_id: int
    area: float | None = None
    occupancy: int | None = None


class RoomUpdate(BaseModel):
    room_name: str | None = None
    room_number: str | None = None
    room_type: str | None = None
    area: float | None = None
    occupancy: int | None = None


class RoomResponse(BaseModel):
    id: int
    room_name: str
    room_number: str | None
    room_type: str | None
    floor_id: int
    area: float | None
    occupancy: int | None

    model_config = ConfigDict(from_attributes=True)
